### ISSUE TITLE
There is a problem with connecting to your Elasticsearch host error

### ISSUE THREAD

**Issue by scooterlord** [10up/ElasticPress#1793]
**Describe the bug**
I encounter an error that reads:
There is a problem with connecting to your Elasticsearch host. ElasticPress can try your host again, or you may need to change your settings. Response Code: 200 Response error: OK

I have searched multiple topics concerning the same issue here at github, but none of them provided a solution.

**Steps to Reproduce**
1. I run elasticsearch server and setup elasticpress
2. I can hit elasticsearch on my browser
3. I can see the instance running in Index Health
4. I can't access the Elasticpress dashboard due to the error... everything is greyed-out.

**Expected behavior**
I would expect to have access to the dashboard.

**Screenshots**
![Screenshot_2020-06-02 ElasticPress ‹ WP Elastic — WordPress](https://user-images.githubusercontent.com/5257454/83535638-df19a300-a4fa-11ea-91c9-ffc2bbb3ef6a.png)

**Environment information**
 - Device: Desktop
 - OS: Windows 10
 - Browser and version: Firefox 76.0.1 64-bit
 - WordPress version: 5.4.1
 - Plugins and version: no plugins other than Elasticpress
 - Theme and version: Twenty Twenty
 
 - <details><summary>Site Health Info:</summary>
`
### wp-core ###

version: 5.4.1
site_language: en_US
user_language: en_US
timezone: +00:00
permalink: /%year%/%monthnum%/%day%/%postname%/
https_status: false
user_registration: 0
default_comment_status: open
multisite: false
user_count: 1
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: C:\UniServerZ\www\wpelastic
wordpress_size: 39.44 MB (41358583 bytes)
uploads_path: C:\UniServerZ\www\wpelastic/wp-content/uploads
uploads_size: 0.00 B (0 bytes)
themes_path: C:\UniServerZ\www\wpelastic/wp-content/themes
themes_size: 3.93 MB (4121752 bytes)
plugins_path: C:\UniServerZ\www\wpelastic/wp-content/plugins
plugins_size: 7.28 MB (7636872 bytes)
database_size: 2.13 MB (2228224 bytes)
total_size: 52.78 MB (55345431 bytes)

### wp-active-theme ###

name: Twenty Twenty (twentytwenty)
version: 1.3
author: the WordPress team
author_website: https://wordpress.org/
parent_theme: none
theme_features: automatic-feed-links, custom-background, post-thumbnails, custom-logo, title-tag, html5, align-wide, responsive-embeds, customize-selective-refresh-widgets, editor-color-palette, editor-font-sizes, editor-styles, widgets, menus, editor-style
theme_path: C:\UniServerZ\www\wpelastic/wp-content/themes/twentytwenty

### wp-themes-inactive (2) ###

Twenty Nineteen: version: 1.5, author: the WordPress team
Twenty Seventeen: version: 2.3, author: the WordPress team

### wp-plugins-active (3) ###

Debug Bar: version: 1.0.1, author: wordpressdotorg
Debug Bar ElasticPress: version: 1.4, author: 10up
ElasticPress: version: 3.4.1, author: 10up

### wp-plugins-inactive (2) ###

Akismet Anti-Spam: version: 4.1.5, author: Automattic
Hello Dolly: version: 1.7.2, author: Matt Mullenweg

### wp-media ###

image_editor: WP_Image_Editor_GD
imagick_module_version: Not available
imagemagick_version: Not available
gd_version: bundled (2.1.0 compatible)
ghostscript_version: not available

### wp-server ###

server_architecture: Windows NT 10.0 i586
httpd_software: Apache
php_version: 7.1.1
php_sapi: apache2handler
max_input_variables: 1000
time_limit: 300
memory_limit: 256M
max_input_time: 60
upload_max_size: 200M
php_post_max_size: 200M
curl_version: not available
suhosin: false
imagick_availability: false
htaccess_extra_rules: false

### wp-database ###

extension: mysqli
server_version: 5.6.35
client_version: mysqlnd 5.0.12-dev - 20150407 - $Id: b396954eeb2d1d9ed7902b8bae237b287f21ad9e $

### wp-constants ###

WP_HOME: undefined
WP_SITEURL: undefined
WP_CONTENT_DIR: C:\UniServerZ\www\wpelastic/wp-content
WP_PLUGIN_DIR: C:\UniServerZ\www\wpelastic/wp-content/plugins
WP_MAX_MEMORY_LIMIT: 256M
WP_DEBUG: true
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_LOCAL_DEV: undefined
DB_CHARSET: utf8mb4
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable

`
   </details>

**Additional context**
Well, that's about it. Elasticpress seems to recognize elasticpress, but for some reason it's not working :/
edit: Don't know if it's any help, but enabling Elasticpress 'hides' my post from the index. Disabling it restores them.

**Comment by scooterlord** [comment]
I reinstalled everything - including my hosting service - and everything seems to be working properly now.

**Comment by scooterlord** [comment]
New edit: I reinstalled everything from scratch, including the local server.

I have combined the following solutions, which only solve the problem temporarily, but thankfully it's reproducible.

a) I added in the `functions.php` the following snippet:
`add_filter( 'ep_format_request_headers', function ( $headers ) {
    $headers['Accept-Encoding'] = 'gzip';
    return $headers;
});`

b) I added in the `wp-config.php` the following:
`define( 'EP_HOST', 'http://127.0.0.1:9200' );`

Now here is the strange part. If I remove the port from the above line like `http://127.0.0.1` I get a different error - that my Elasticsearch version is below the minimum required Elasticsearch version 5.0. ElasticPress may or may not work properly. If I restore the port back in the `wp-config` then the Dashboard works only ONCE, until I navigate elsewhere or refresh the page and then fall back to the same issue  - There is a problem with connecting to your Elasticsearch host. ElasticPress can try your host again, or you may need to change your settings. 

Hopefully this information helps.

**Comment by felipeelia** [comment]
Hi @scooterlord ,

ElasticPress tests the Elasticsearch server making a request to the `/_nodes/plugins` endpoint.

Are you able to visit `http://127.0.0.1:9200/_nodes/plugins`? If so, do you get a JSON object with a "nodes" attribute?

And the port part is expected. As I said, ElasticPress will try to find the Elasticsearch info through an HTTP GET request. If you set the host as `http://127.0.0.1` you're pointing it to your web server (Apache or Nginx, for example). Although the script will receive an answer, that will not contain the info Elasticsearch would return.

**Comment by scooterlord** [comment]
Hello @felipeelia and thanks for your reply,

I realize that it required the port, but changing back and forth without and with port was what enabled the plugin dashboard to be visible once.

I came here to post the solution to the problem, so checking `http://127.0.0.1:9200/_nodes/plugins` _now finally works_ as it should, 

After multiple reinstallations of the server, wordpress and elasticpress I found what the problem was. I had seen the fix to the problem elsewhere but ignored it, since I took it for granted. I am using the Uniform Server for my local server that has the `curl` module disabled by default. I hope this helps people arriving in this post.
